### PR TITLE
DRi#1557: DR CMake functions now take extra arg

### PR DESCRIPTION
**Pull request by hgreving2304**
Adding location suffix parameter to calls of DynamoRIO_get_target_path_for_execution, DynamoRIO_copy_target_to_device. This patch does not address the LOCATION part of i#1557 itself.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1050,7 +1050,7 @@ endif ()
```
Comment (derekbruening) [inline CMakeLists.txt]: I don't see the dynamorio submodule being updated?  So this is passing extra args to functions that don't take them?  I guess CMake doesn't complain and things pass but that's not what we want.  We want to update the submodule.

Comment (derekbruening) [inline CMakeLists.txt]: > i#1557: compatibility update, CMake API dynamorio has changed.

The issue namespaces are different: 1557 is a different issue here.  See prior issues: they use `DRi#1557` to refer to a DR issue in a title.  DynamoRIO/dynamorio/issues/1557 will auto-link in the body.

Comment (derekbruening) [inline CMakeLists.txt]: > i#1557: compatibility update, CMake API dynamorio has changed.

grammar nit: comma isn't right there: should be a colon or sthg.  suggest sthg like: `DRi#1557: DR CMake functions now take extra arg` which gives an idea of what changed.

Comment (hgreving2304) [reply CMakeLists.txt]: Done

Comment (hgreving2304) [reply CMakeLists.txt]: I think I just did, please double-check if that was right (just git added dynamorio to latest)

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -500,7 +500,7 @@ newtest_ex(state state.c "" "" "" OFF "" "ANY")
 if (UNIX)
   newtest(signal signal.c)
   if (NOT X64) # FIXME i#111: failing on Travis
-    newtest(syscalls_unix syscalls_unix.c)
+    #newtest(syscalls_unix syscalls_unix.c)
```
Comment (derekbruening) [inline tests/CMakeLists.txt]: Please examine the diff before pushing

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -536,7 +536,7 @@ if (UNIX)
           VERBATIM)
       add_custom_target("${kmod_name}"
           DEPENDS "${kmod_bin}")
-      add_dependencies(syscalls_unix "${kmod_name}")
+      #add_dependencies(syscalls_unix "${kmod_name}")
```
Comment (derekbruening) [inline tests/CMakeLists.txt]: Another mistakenly included change.  Is this that #2049?  Is it just missing `if (NOT X64)?`

File: tests/CMakeLists.txt (hunk 2 above)
Comment (hgreving2304) [reply tests/CMakeLists.txt]: Done

File: tests/CMakeLists.txt (hunk 3 above)
Comment (hgreving2304) [reply tests/CMakeLists.txt]: Done

Comment (hgreving2304) [reply tests/CMakeLists.txt]: Yes it might, I had changed this locally to be able to build.

**Pull request merged**